src/Components/VideoModal/VideoModal.tsx

```diff
@@ -0,0 +1,222 @@
+import React, { useEffect, useState } from "react";
+import { Button, Modal, Container, Carousel } from "react-bootstrap";
+import { VideoProgressBar } from "./VideoProgressBar";
+import { templates, templateEl, slotEl } from "../Template";
+import styles from "./VideoModal.module.css";
+import { MediaComponent } from "./MediaComponent";
+import RotateLoader from "react-spinners/RotateLoader";
+import { MediaPresenter } from "../MediaPresenter";
+
+type Props = {
+  setShow: (show: boolean) => void;
+  show: boolean;
+  mediaPresenter: MediaPresenter;
+};
+export const VideoModal: React.FC<Props> = ({
+  setShow,
+  show,
+  mediaPresenter
+}) => {
+  const [mediaCounter, setMediaCounter] = useState<number>(0);
+  const [shuffledCounter, setShuffledCounter] = useState<number[]>(
+    mediaPresenter.shuffleArray()
+  );
+  // default template is neutral, short
+  const [currTemplate, setCurrTemplate] = useState<templateEl>(
+    templates.neutral
+  );
+  const [music, setMusic] = useState<HTMLAudioElement>(
+    new Audio(currTemplate.musicTrack)
+  );
+  const [musicLoaded, setMusicLoaded] = useState<boolean>(false);
+  const [length, setLength] = useState<slotEl>(currTemplate.medium);
+  const [index, setIndex] = useState(0);
```

Comment:
This is a lot of state!
What of this could live inside your `mediaPresenter` instead?
Or, can you create another presenter to handle everything output related - eg `outputPresenter`?

Reply:
I have put everything inside mediaPresenter :D